### PR TITLE
BLD: Pinned cftime 1.0.1

### PR DESCRIPTION
**Pull request by cpelley**
#3239, #3215, #3225

### CODE REVIEW

**Comment by rcomer** [comment]
My user code is failing with cftime version 1.0.0b1 so I am not sure if `<1.0.2.1` will solve all problems (although getting Travis checks to pass is obviously desirable!)  What version would Travis be picking up here?  I'm sure that info must be in the output somewhere but I can't see it!

**Comment by cpelley** [comment]
Travis is using cftime 1.0.1 (conda-forge)

**Comment by pp-mo** [comment]
>  cftime version 1.0.0b1 ... What version would Travis be picking up here? I'm sure that info must be in the output somewhere but I can't see it!

You can find this from a Travis log : 
To view a log, first click on the tests icon by the commit in Github (green tick or red cross!),
navigate to the "continuous-integration/travis-ci/pr ... Details". 
Hit another tick (or cross) for the log of one testing "phase" 
(e.g. python 2.7, TEST_TARGET=default, TEST_MINIMAL=True). 

In the log, look for the line "$ conda list -n $ENV_NAME ....",
the initial wedge ">" opens the collapsed command output ...
which is the list of installed packages !!

In this case, "cftime 1.0.1"

**Comment by rcomer** [comment]
Thanks both.  I hadn't even twigged that there was collapsed output in the Travis log!

File: requirements/core.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@
 cartopy
 #conda: proj4<5
 cf_units>=2
-cftime!=1.0.2.1
+cftime<1.0.2.1
```
Comment (bjlittle) [inline requirements/core.txt]: Why not pin directly to `1.0.1`, just to make this explicit and clear.

We still need to fix `iris` to deal with 0 based dates, which are now illegal in the latest `cftime`.

So this pin would only be a temporary pin...

Comment (cpelley) [reply requirements/core.txt]: > ...Why not pin directly...

Sure.

> We still need to fix iris to deal with 0 based dates...

Agreed.  Some relating discussion in #3239

**Comment by bjlittle** [comment]
@cpelley Apologies. Sorry to be a pain...

But could you please target the `2.2.x` branch instead of `master`, and then I'll merge back the `2.2.x` branch to `master`...

**Comment by cpelley** [comment]
NP :)

File: requirements/core.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@
 cartopy
 #conda: proj4<5
 cf_units>=2
-cftime!=1.0.2.1
+cftime>=1.0.1,<=1.0.1
```
Comment (bjlittle) [inline requirements/core.txt]: @cpelley Surely `cftime==1.0.1` is what you mean, right? Doesn't that work?

**Comment by bjlittle** [comment]
Thanks @cpelley :+1: 

Ping @pp-mo

**Pull request merged**